onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1081,6 +1124,10 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BeamSearch, 1,
                                       "Size of the vocabulary. "
                                       "If not provided, it will be inferred from the decoder subgraph's output shape",
                                       AttributeProto::INT, static_cast<int64_t>(-1))
+                                .Attr("decoder_output_cross_qk", "If nozero, decoder subgraph contains output Q*K from cross attentions. Default 0.", AttributeProto::INT, OPTIONAL_VALUE)
```

Comment:
Since it need so many changes in BeamSearch interface, I suggest to add a new operator.

Reply:
Yeah. worth to make a separate op for this.